src/fsharp/AugmentWithHashCompare.fs

```diff
@@ -626,27 +626,36 @@ let mkUnionHashWithComparer g tcref (tycon:Tycon) compe =
     let mkCase i ucase1 = 
         let c1ref = tcref.MakeNestedUnionCaseRef ucase1 
         let m = c1ref.Range 
-        let mkHash thise j (rfield:RecdField) =  
-            let fty = rfield.FormalType
-            let e = mkUnionCaseFieldGetProvenViaExprAddr(thise, c1ref, tinst, j, m)
-            mkCallGenericHashWithComparerOuter g m fty compe e
-
-        let test =       
-            if tycon.IsStructOrEnumTycon then 
-                mkCompGenSequential m 
-                    (mkValSet m (mkLocalValRef accv) (mkInt g m i)) 
-                    (mkCombineHashGenerators g m (List.mapi (mkHash thise) ucase1.RecdFields) (mkLocalValRef accv) acce)
-            else
-                let ucv,ucve = mkCompGenLocal m "unionCase" (mkProvenUnionCaseTy c1ref tinst)
-                mkCompGenLet m ucv
-                    (mkUnionCaseProof (thise,c1ref,tinst,m))
-                    (mkCompGenSequential m 
+        if ucase1.IsNullary then None 
+        else
+            let mkHash thise j (rfield:RecdField) =  
+                let fty = rfield.FormalType
+                let e = mkUnionCaseFieldGetProvenViaExprAddr(thise, c1ref, tinst, j, m)
+                mkCallGenericHashWithComparerOuter g m fty compe e
+
+            let test =       
+                if tycon.IsStructOrEnumTycon then 
+                    mkCompGenSequential m 
                         (mkValSet m (mkLocalValRef accv) (mkInt g m i)) 
-                        (mkCombineHashGenerators g m (List.mapi (mkHash ucve) ucase1.RecdFields) (mkLocalValRef accv) acce))
-
-        mkCase(Test.UnionCase(c1ref,tinst),mbuilder.AddResultTarget(test,SuppressSequencePointAtTarget))
-
-    let dtree = TDSwitch(thise, List.mapi mkCase ucases, None,m) 
+                        (mkCombineHashGenerators g m (List.mapi (mkHash thise) ucase1.RecdFields) (mkLocalValRef accv) acce)
+                else
+                    let ucv,ucve = mkCompGenLocal m "unionCase" (mkProvenUnionCaseTy c1ref tinst)
+                    mkCompGenLet m ucv
+                        (mkUnionCaseProof (thise,c1ref,tinst,m))
+                        (mkCompGenSequential m 
+                            (mkValSet m (mkLocalValRef accv) (mkInt g m i)) 
+                            (mkCombineHashGenerators g m (List.mapi (mkHash ucve) ucase1.RecdFields) (mkLocalValRef accv) acce))
+            Some(mkCase(Test.UnionCase(c1ref,tinst),mbuilder.AddResultTarget(test,SuppressSequencePointAtTarget)))
+
+    let nullary,nonNullary = ucases
```

Comment:
Thanks. I'll try to remember to do so in the future. One quick clarification though: I assume it's alright if there are enough pipes in a row to put pipes on their own lines?

Also, do you want me to fix it in this PR?